src/protocols/vnc/auth.c

```diff
@@ -31,3 +31,19 @@ char* guac_vnc_get_password(rfbClient* client) {
     return ((guac_vnc_client*) gc->data)->settings->password;
 }
 
+rfbCredential* guac_vnc_get_credentials(rfbClient* client, int credentialType) {
+    guac_client* gc = rfbClientGetClientData(client, GUAC_VNC_CLIENT_KEY);
+    guac_vnc_settings* settings = ((guac_vnc_client*) gc->data)->settings;
+    
+    if (credentialType == rfbCredentialTypeUser) {
+        rfbCredential *creds = malloc(sizeof(rfbCredential));
+        creds->userCredential.username = settings->username;
+        creds->userCredential.password = settings->password;
+        return creds;
+    }
+    
+    guac_client_log(gc, GUAC_LOG_ERROR,
+            "Unsupported credential type requested.");
```

Comment:
I haven't looked into the behavior of libvncclient with respect to this callback. Is a request for an unsupported credential type an error? Or is it something which is expected to occur when a VNC server supports multiple authentication mechanisms, not all of which are necessarily supported by the client?

Reply:
I believe for the callback function it is an error, because this will be called when the client detects the need for additional credentials, and, if the requested credentials are not provided, this constitutes an error that the client cannot handle - that is, the client cannot fulfill the required credentials.

Does that make sense?

Reply:
Sure, makes perfect sense.

In that case, I would recommend:

1. Using `guac_client_abort()`, so this is explicitly fatal to the connection.
2. Additionally logging the credential type that's unsupported, at least at the debug level.

Reply:
Okay, I think I did this correctly.  There doesn't seem to be an easy way to go from the `int` type requested credential back to some sort of human-readable name, so, while that's less-than-ideal, hopefully that's okay...